File: ISAMedicalEquipmentFrontend/src/app/domain/user/system-administrator.ts
import { DateTime } from "luxon";

import { User } from "./user";
import { UserRole } from "./user-role";

export class SystemAdministrator extends User {
    employedSince: DateTime;

    constructor(id: number, roles: UserRole[], isEnabled: boolean, userCode: string, emailAddress: string, 
            username: string, password: string | null, lastPasswordResetDate: number | DateTime | null, 
            firstName: string, lastName: string, residence: string, populatedPlace: string, country: string, 
            phoneNumber: string, personalIdentityNumber: string, gender: string, profession: string | null, 
            companyName: string | null, employedSince: number | DateTime) {
        super(id, roles, isEnabled, userCode, emailAddress, username, password, lastPasswordResetDate, firstName, 
                lastName, residence, populatedPlace, country, phoneNumber, personalIdentityNumber, gender, profession, 
                companyName);

        if (DateTime.isDateTime(employedSince)) {
            this.employedSince = employedSince;
        } else {
            this.employedSince = DateTime.fromMillis(employedSince);
        }
    }
}

File: ISAMedicalEquipmentBackend/src/main/java/ftn/project/ISAMedicalEquipmentBackend/domain/user/SystemAdministrator.java
package ftn.project.ISAMedicalEquipmentBackend.domain.user;

import java.sql.Timestamp;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

@Entity
@Table(name = "system_administrators")
public class SystemAdministrator extends User {
	@Column(name = "employed_since", nullable = false)
	private Timestamp employedSince;
	
	public SystemAdministrator() {}
	
	public SystemAdministrator(long id, Set<UserRole> roles, boolean isEnabled, String userCode, 
			String emailAddress, String username, String password, Timestamp lastPasswordResetDate, 
			String firstName, String lastName, String residence, String populatedPlace, 
			String country, String phoneNumber, String personalIdentityNumber, Gender gender, 
			String profession, String companyName, double companyLatitude, double companyLongitude, 
			Timestamp employedSince) {
		super(id, roles, isEnabled, userCode, emailAddress, username, password, 
				lastPasswordResetDate, firstName, lastName, residence, populatedPlace, country, 
				phoneNumber, personalIdentityNumber, gender, profession, companyName, 
				companyLatitude, companyLongitude);
		
		this.employedSince = employedSince;
	}
	
	public Timestamp getEmployedSince() {
		return employedSince;
	}
	
	public void setEmployedSince(Timestamp employedSince) {
		this.employedSince = employedSince;
	}
}
